cpp/src/arrow/util/bit-util.h

```diff
@@ -570,6 +570,18 @@ ARROW_EXPORT
 Status CopyBitmap(MemoryPool* pool, const uint8_t* bitmap, int64_t offset, int64_t length,
                   std::shared_ptr<Buffer>* out);
 
+/// Copy a bit range of an existing bitmap into an existing bitmap
+///
+/// \param[in] bitmap source data
+/// \param[in] offset bit offset into the source data
+/// \param[in] length number of bits to copy
+/// \param[in] dest_offset bit offset into the destination
+/// \param[out] dest the destination buffer, must have at least space for (offset +
+/// length) bits
+ARROW_EXPORT
+void CopyBitmap(const uint8_t* bitmap, int64_t offset, int64_t length,
```

Comment:
The signature is a bit irregular, I would expect `(const uint8_t* bitmap, int64_t offset, int64_t length, uint8_t* dest, int64_t dest_offset)` or something.